### PR TITLE
issue 435.1: allow a user to specify the number of IResearchFeature threads via command line

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/619/

File: arangod/IResearch/IResearchFeature.cpp [hunk 1]
```diff
@@ -144,6 +144,20 @@ typedef arangodb::aql::AqlValue (*IResearchFunctionPtr)(
...
 );
 
+size_t computeThreadPoolSize(size_t threads, uint32_t threadsLimit) {
+  static const unsigned int MAX_THREADS = 8; // arbitrary limit on the upper bound of threads in pool
+  static const unsigned int MIN_THREADS = 1; // at least one thread is required
+  auto maxThreads = threadsLimit ? threadsLimit : MAX_THREADS;
+
+  return threads
+    ? threads
+    : std::max(
```
Comment (jsteemann) [inline arangod/IResearch/IResearchFeature.cpp]: Using different types for `std::min` or `std::max` is like asking for trouble.
At least the compilers on MacOS may be picky about it if you mix `size_t`, `uint32_t` or `unsigned int`. These types may boil down to the same fundamental type or not.
It is safest and easiest to understand to use a single data type for the number of threads (either `size_t`, `uint32_t` or `unsigned int`, but not 3 different ones.

File: arangod/IResearch/IResearchFeature.cpp [hunk 2]
```diff
@@ -632,6 +652,35 @@ void IResearchFeature::Async::start() {
...
+        _join.wait();
+      }
+    }
+
+    // redelegate all thread tasks if requested
+    if (redelegate) {
+      SCOPED_LOCK(redelegate->_mutex);
+
+      for (auto& task: thread._pending) {
+        redelegate->_pending.emplace_back(std::move(task));
```
Comment (jsteemann) [inline arangod/IResearch/IResearchFeature.cpp]: The `emplace_back` may throw an exception, but is called from the destructor...
Should be wrapped in a try..catch block either here or the call in the destructor should be wrapped into try..catch.

Comment (vasiliy-arangodb) [reply arangod/IResearch/IResearchFeature.cpp]: the call from the destructor will have redelegate == nullptr

Comment (jsteemann) [reply arangod/IResearch/IResearchFeature.cpp]: Ok then

**Pull request merged**